fenix/app/src/main/java/org/mozilla/fenix/compose/ThumbnailImage.kt

```diff
@@ -20,40 +20,35 @@ import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.asImageBitmap
 import androidx.compose.ui.graphics.painter.BitmapPainter
 import androidx.compose.ui.layout.ContentScale
-import androidx.compose.ui.platform.LocalDensity
 import androidx.compose.ui.tooling.preview.Preview
-import androidx.compose.ui.unit.Dp
-import androidx.compose.ui.unit.dp
 import kotlinx.coroutines.launch
+import mozilla.components.browser.thumbnails.storage.ThumbnailStorage
 import mozilla.components.concept.base.images.ImageLoadRequest
-import org.mozilla.fenix.components.components
 import org.mozilla.fenix.theme.FirefoxTheme
 
 /**
- * Thumbnail belonging to a [key]. Asynchronously fetches the bitmap from storage.
+ * Thumbnail belonging to a [ImageLoadRequest]. Asynchronously fetches the bitmap from storage.
  *
- * @param key Key used to remember the thumbnail for future compositions.
- * @param size [Dp] size of the thumbnail.
+ * @param request [ImageLoadRequest] used to fetch the thumbnail bitmap.
+ * @param storage [ThumbnailStorage] to obtain tab thumbnail bitmaps from.
  * @param modifier [Modifier] used to draw the image content.
  * @param contentScale [ContentScale] used to draw image content.
  * @param alignment [Alignment] used to draw the image content.
+ * @param fallbackContent The content to display with a thumbnail is unable to be loaded.
  */
 @Composable
 @Suppress("LongParameterList")
 fun ThumbnailImage(
-    key: String,
-    size: Dp,
+    request: ImageLoadRequest,
+    storage: ThumbnailStorage?,
     modifier: Modifier,
     contentScale: ContentScale,
     alignment: Alignment,
     fallbackContent: @Composable () -> Unit,
 ) {
-    if (inComposePreview) {
+    if (storage == null) {
```

Comment:
IMO if this is the only purpose the nullable parameter is serving it would be best to switch back to `inComposePreview`. The nullable parameter communicates that the storage component is optional, which seems like it could lead to unexpected behavior for consumers of the API

Reply:
We discussed this offline, and I am pushing changes to add `inComposePreview` back in, after discovering I can create a dummy `ThumbnailStorage` inside of previews. So the nullability will also go away as a result.